src/Core/Services/VaultTimeoutService.cs

```diff
@@ -88,7 +88,8 @@ public async Task<bool> ShouldTimeoutAsync(string userId = null)
             {
                 return false;
             }
-            if (await IsLockedAsync(userId))
+            var vaultTimeoutAction = await _stateService.GetVaultTimeoutActionAsync(userId);
+            if (vaultTimeoutAction == "lock" && await IsLockedAsync(userId))
```

Comment:
Maybe not for this PR, but I'd very like to change the `VaultTimeoutAction` to be an enum instead of a `string`, it's easier to compare, less error prone, allocates less memory and you can see all of the values in one place.